From 5d8b659ad0e38196055b63ec75df46ab32254051 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Fri, 21 Jul 2017 14:22:12 +0800
Subject: [PATCH] ARM: dts: rockchip: rk3368-sheep add vsel gpio for syr827

Add vsel gpio for syr827, which is PMIC_SLEEP in hardware pcb.

The driver will parse the gpio and use it to indicate the chip
work status, as the following codes:

static int fan53555_is_enabled ()
	if defined vsel_gpio
		gpiod_get_raw_value
	else
		regmap_read

Before the patch, the log shows:
	CPU4: update cpu_capacity 1024
	CPU4: Booted secondary processor [410fd033]
	  cache: parent cpu4 should not be sleeping
	rk3x-i2c ff650000.i2c: timeout, ipd: 0x10, state: 1

Since cpu4 is the base core for cluster1, which need to initialize
cpufreq during boot up, finally call fan53555_is_enabled. But the
i2c is suspended at that time, can't get interrupt until timeout.

Change-Id: I301e95be3b60d2faa456759d88c06cf64c2019ca
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm64/boot/dts/rockchip/rk3368-sheep.dts | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3368-sheep.dts b/arch/arm64/boot/dts/rockchip/rk3368-sheep.dts
index 649c475e12f9..f18b7a883eff 100644
--- a/arch/arm64/boot/dts/rockchip/rk3368-sheep.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3368-sheep.dts
@@ -221,6 +221,8 @@
 		regulator-max-microvolt = <1500000>;
 		regulator-ramp-delay = <1000>;
 		fcs,suspend-voltage-selector = <1>;
+		pinctrl-0 = <&vsel_gpio>;
+		vsel-gpios = <&gpio0 0 GPIO_ACTIVE_HIGH>;
 		regulator-always-on;
 		regulator-boot-on;
 		regulator-initial-state = <3>;
@@ -645,6 +647,9 @@
 		pmic_int_l: pmic-int-l {
 			rockchip,pins = <RK_GPIO0 1 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
+		vsel_gpio: vsel-gpio {
+			rockchip,pins = <0 0 RK_FUNC_GPIO &pcfg_pull_down>;
+		};
 	};
 
 	dc_det {
-- 
2.35.3

